### ISSUE TITLE
kubernetes deployment to non-default namespace

### ISSUE THREAD

**Issue by nejec** [sorintlab/stolon#452]
We are seeing some strange errors, when we are trying to deploy stolon, using kubernetes example, to different namespace than `default`. The issues arised within `stolon-sentinel` part.

We have added the Role and RoleBinding, used within kubernetes example, however this did not fix the issues.

### Submission type

 - Bug report
### Environment
kubernetes - v1.9.2
Namespace: `stolon` (not `default`)
3 x stolon-proxy
3 x stolon-sentinel
2 x stolon-keeper

#### Stolon version
stolon - v0.9.0

#### Additional environment information if useful to understand the bug
Deployed with kubernetes example, available here: https://github.com/sorintlab/stolon/tree/master/examples/kubernetes

### Unexpected behaviour you saw

The logs from `stolon-sentinel`. Every `stolon-sentinel` pod reports the same error, described below.
```
ERROR: logging before flag.Parse: E0301 14:59:12.694568       1 event.go:209] Unable to write event: ’can’t create an event with namespace ‘stolon’ in namespace ‘default’' (may retry after sleeping)
ERROR: logging before flag.Parse: E0301 14:59:12.694786       1 event.go:144] Unable to write event ‘&amp;v1.Event{TypeMeta:v1.TypeMeta{Kind:“”, APIVersion:“”}, ObjectMeta:v1.ObjectMeta{Name:“stolon-cluster-kube-stolon.1517d3c5638ace15”, GenerateName:“”, Namespace:“stolon”, SelfLink:“”, UID:“”, ResourceVersion:“”, Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:“”}, InvolvedObject:v1.ObjectReference{Kind:“ConfigMap”, Namespace:“stolon”, Name:“stolon-cluster-kube-stolon”, UID:“9bc5cb46-1d5c-11e8-8cc9-025000000001”, APIVersion:“v1", ResourceVersion:“35552”, FieldPath:“”}, Reason:“LeaderElection”, Message:“de271f34 became leader”, Source:v1.EventSource{Component:“stolon-sentinel”, Host:“”}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xbe9e2314fb2e5015, ext:18945899077, loc:(*time.Location)(0x1e76d60)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xbe9e2314fb2e5015, ext:18945899077, loc:(*time.Location)(0x1e76d60)}}, Count:1, Type:“Normal”}’ (retry limit exceeded!)
```
### Steps to reproduce the problem

Change the `default` namespace to any other, already created empty namespace in k8s.

**Comment by sgotti** [comment]
@nejec Are you using master right (since the k8s store was merged few days ago)?

I also noticed that error last day but it was harmful and the sentinel should work anyway (it''s reported by the k8s client leader election code).

It'll need a fix to the rbac permissions to let the sentinel also wrote events.

**Comment by nejec** [comment]
@sgotti we are using current `master-pg9.6` docker image

And yes, we noticed that cluster works, however we were not sure, whether this could impact the whole system.

btw: thanks for really fast reply ;)

**Comment by nejec** [comment]
by using role definitions below, we are seeing the same errors...

```{{- if .Values.rbac.create -}}
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: Role
metadata:
  name: {{ template "stolon.fullname" . }}
  namespace: {{ .Release.Namespace | quote }}
  labels:
    app: {{ template "stolon.name" . }}
    chart: {{ template "stolon.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
  annotations:
    "helm.sh/hook": pre-install
rules:
  - apiGroups:
    - ""
    resources:
      - configmaps
      - deployments
      - endpoints
      - events
      - jobs
      - pods
      - persistentvolumes
      - persistentvolumeclaims
      - statefulsets
      - secrets
      - services
    verbs:
      - "*"

---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: RoleBinding
metadata:
  name: {{ template "stolon.fullname" . }}
  namespace: {{ .Release.Namespace | quote }}
  annotations:
    "helm.sh/hook": pre-install

roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: {{ template "stolon.fullname" . }}
subjects:
- kind: ServiceAccount
  name: {{ template "stolon.serviceAccountName" . }}
  namespace: {{ .Release.Namespace | quote}}

{{- end -}}```

**Comment by sgotti** [comment]
Created PR #453

**Comment by sgotti** [comment]
#Fixed in #453
